tests/c/nested_sidetrace.c

```diff
@@ -32,16 +32,17 @@
 //     18
 //     yk-jit-event: enter-jit-code
 //     yk-jit-event: deoptimise
-//     yk-jit-event: start-side-tracing
```

Comment:
Why has the PR changed when side tracing starts?

Reply:
Because the guard in the header and the guard in the body are two different guards. So we typically fail one iteration later than before (in order to hit 5 guard failures we need to fail 6 times since the first failure is in the body). I tried to minimise the changes by reducing the default threshold to 4, but that didn't do the trick for all tests (likely because not all tests have the header guard fail immediately).

Reply:
Ah, right, that makes sense. Let's put the default threshold back to 5: it's better to just bite the bullet and adjust the tests now. If you want -- but I wouldn't necessarily do this -- you can set the threshold to 4 in individual tests.

Reply:
Ok, but note this will implicitly increase the side-trace threshold to 6, at least as far as an external observer is concerned.

Reply:
Hmm, yeah, we probably need to fix this more thoroughly. OK, let's do the `4` hack for now.

Reply:
[0ae874b](https://github.com/ykjit/yk/pull/1510/commits/0ae874b880bd85495f277e63bde4c15e43e245bb)

I've reset it now. :-/ Do you want me to revert it again?

I've also left the manual adjustment in the `side-trace.c` test in as it was easier this way.

Reply:
Do whichever is easiest for now.

Reply:
> Hmm, yeah, we probably need to fix this more thoroughly. OK, let's do the `4` hack for now.

Yes, also since a side-trace itself doesn't have a header and thus setting a threshold of 5 will behave as expected.

Reply:
Ok, let's leave it how it is now.